### DISCUSSION TITLE
Flatten form fields

### DISCUSSION THREAD

**Discussion by Flowerinno** [Hopding/pdf-lib#1485] [Q&A]
I have a pdf document with form fields , after setting form values I use form.flatten() and save the document , in result I have a valid pdf file , but I need to send this file via Pengin mail and it says that the file is invalid or corrupt.

```
 const pdfDoc = await PDFDocument.load(existingPdfBytes, {
      updateMetadata: true,
    });

    const form = pdfDoc.getForm();
    /// get fields and set values
    
    
     form.flatten();

    const pdfBytes = await pdfDoc.save();
    ```

**Comment by Jeff-Hamby** [comment]
I'm getting the same thing, I did commit out the form.flatten() and it seemed to work.  But that leaves the form open to editing, which I don't want.
